ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BodyValidation.java

```diff
@@ -97,14 +97,7 @@ public static Hash withdrawalsRoot(final List<Withdrawal> withdrawals) {
    */
   public static Hash requestsHash(final List<Request> requests) {
     List<Bytes> requestHashes = new ArrayList<>();
-    requests.forEach(
-        request -> {
-          // empty requests are excluded from the hash
-          if (!request.getData().isEmpty()) {
-            requestHashes.add(sha256(request.getEncodedRequest()));
-          }
```

Comment:
I think we still need to filter out the empty requests when we are building our blocks

Reply:
Done